src/dsp_tools/utils/xmlupload/xmlupload.py

```diff
@@ -352,93 +332,81 @@ def _upload_resources(
-            except BaseError as err:
-                pth = resource.bitstream.value
-                err_msg = err.orig_err_msg_from_api or err.message
-                msg = f"Unable to upload file '{pth}' of resource '{resource.label}' ({resource.id})"
-                print(f"WARNING: {msg}: {err_msg}")
-                logger.warning(msg, exc_info=True)
-                msg = f"Uploaded file '{pth}' ({bitstream_size_uploaded_mb:.1f} MB / {bitstream_size_total_mb} MB)"
-                print(msg)
-                logger.info(msg)
-                continue
-        else:
-            resource_bitstream = None
-
-        # create the resource in DSP
-        resclass_type = resclass_name_2_type[resource.restype]
-        properties = resource.get_propvals(id2iri_mapping, permissions_lookup)
-        try:
-            resource_instance: ResourceInstance = resclass_type(
-                con=con,
-                label=resource.label,
-                iri=resource_iri,
-                permissions=permissions_lookup.get(str(resource.permissions)),
-                creation_date=resource.creation_date,
-                bitstream=resource_bitstream,
-                values=properties,
-            )
-            resource_creation_start = datetime.now()
-            created_resource: ResourceInstance = try_network_action(resource_instance.create)
-            resource_creation_duration = datetime.now() - resource_creation_start
-            resource_creation_duration_ms = resource_creation_duration.seconds * 1000 + int(
-                resource_creation_duration.microseconds / 1000
-            )
-            metrics.append(
-                MetricRecord(resource.id, filetype, filesize, "resource creation", resource_creation_duration_ms, "")
+        bitstream_information = None
+        if bitstream := resource.bitstream:
+            bitstream_information = handle_bitstream(
+                resource=resource,
+                bitstream=bitstream,
+                preprocessing_done=preprocessing_done,
+                permissions_lookup=permissions_lookup,
+                sipi_server=sipi_server,
+                imgdir=imgdir,
             )
-        except BaseError as err:
-            err_msg = err.orig_err_msg_from_api or err.message
-            print(f"WARNING: Unable to create resource '{resource.label}' ({resource.id}): {err_msg}")
-            log_msg = (
-                f"Unable to create resource '{resource.label}' ({resource.id})\n"
-                f"Resource details:\n{vars(resource)}\n"
-                f"Property details:\n" + "\n".join([str(vars(prop)) for prop in resource.properties])
-            )
-            logger.warning(log_msg, exc_info=True)
+            if not bitstream_information:
+                # Note: previously, if the bitstream could not be uploaded, the resource was skipped without adding to
+                # the failed_uploads list, which I'm not sure was as intended
```

Comment:
@jnussbaum what do you think?

Reply:
No, that was not intended. Thanks that you caught this!

Reply:
Ok, in that case I'll leave it as it is now and just remove the comment. Thanks for the clarification

Reply:
But now I realize that the old lines 390-392 (https://github.com/dasch-swiss/dsp-tools/blob/03d7fca22b7d30f232bae5f09e5a93439420f675/src/dsp_tools/utils/xmlupload/xmlupload.py#L390) had a wrong indentation! Did you notice that, and correct it in your PR?

Reply:
If I see correctly, the resource is silently skipped if the file cannot be uploaded. I don't think that's ideal...